metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiRecordParser.java

```diff
@@ -79,58 +81,48 @@ private static void initializeExpressions() throws HarvesterException {
    * @param oaiRecord The record to parse as a string value.
    * @throws HarvesterException In case there was a problem with setting up the parser.
    */
-  OaiRecordParser(String oaiRecord) throws HarvesterException {
+  OaiRecordParser(byte[] oaiRecord) throws HarvesterException {
```

Comment:
The oaiRecord shouldn't be part of the class or constructor.
It should be used/passed with the methods further down, right?
Also the XPathExpression doesn't have to be computed for every record, just once. Computing them each time I believe has a significant impact on the performance.

Reply:
True, but at least we avoid that extra call to the initialization